pkg/loki/topology_query.go

```diff
@@ -49,19 +47,39 @@ func NewTopologyQuery(cfg *Config, start, end, limit, reporter string) (*Topolog
 		}
 	}
 
+	var f string
+	switch metricFunction {
+	case "avg":
+		f = "avg_over_time"
+	case "max":
+		f = "max_over_time"
+	case "rate":
+		f = "rate"
+	default:
+		f = "sum_over_time"
+	}
+
+	var t string
+	switch metricType {
+	case "packets":
+		t = "Packets"
+	default:
+		t = "Bytes"
+	}
+
 	return &TopologyQueryBuilder{
 		FlowQueryBuilder: NewFlowQueryBuilder(cfg, start, end, limit, reporter),
 		topology: &Topology{
-			timeRange: timeRange,
```

Comment:
Interesting. On my side I get the correct numbers like `7908/min` for `482000` in total.

The query is built the following way:

```
<url path>?query=
	topk(
	 	<k>,
		sum by(<aggregations>) (
			<function>(
				{<label filters>}|<line filters>|json|<json filters>
					|unwrap Bytes|__error__=""[<time>s]
			)
		)
	)
	&<query params>&step=60s
```

So you will get a set of values (one per minute) of `average` or `sum` in your example. Maybe network sampling makes avg function wrong since it aggregate flows in a single time :thinking: 

Also we can play with the step:
```go
//TODO: check if step should be configurable. 60s is forced to help calculations on front end side
sb.WriteString("&step=60s")
```

Reply:
As seen together, this appear when netflows are not communicating during a `60s` period of time (step). In that case Loki doesn't return any value for this time range and the avg / rate total calculation is false.

I have pushed a fix. Thanks @OlivierCazade :1st_place_medal: